torch/distributed/_shard/sharded_tensor/_ops/matrix_ops.py

```diff
@@ -0,0 +1,67 @@
+import copy
+
+import torch
+from torch.distributed._shard.sharded_tensor import (
+    Shard,
+    ShardedTensor,
+)
+
+from ._common import (
+    _register_sharded_op_on_local_shards,
+)
+
+
+def sharded_type_as_check(*args, **kwargs):
+    """
+    Perform extra checks for the sharded_type_as op such as the input needs to
+    be either a Tensor or ShardedTensor.
+
+    Args: same as ``torch.Tensor.type_as``.
+
+    Return: None
+    """
+    if len(args) < 2:
+        raise ValueError("Needs to give a tensor to cast type as!")
+    if not isinstance(args[1], torch.Tensor) and not isinstance(args[1], ShardedTensor):
+        raise ValueError("Needs to give a Tensor or ShardedTensor to cast type as!")
+
+
+def same_dtype(*args, **kwargs):
```

Comment:
Weren't these already added in https://github.com/pytorch/pytorch/pull/75374?

Reply:
Yep. It somehow gets removed during rebase? I also felt wired about `type_as`. Originally the test fails so I have to add them back.